src/ICSharpCode.SharpZipLib/Zip/ZipStrings.cs

```diff
@@ -20,175 +25,100 @@ static ZipStrings()
+				? throw new ArgumentOutOfRangeException(nameof(value))
+				: Encoding.GetEncoding(value);
+		}
+
+		private const int FallbackCodePage = 437;
 
 		/// <summary>
-		/// Convert a byte array to a string  using <see cref="CodePage"/>
+		/// Operating system default codepage, or if it could not be retrieved, the fallback code page IBM 437.
 		/// </summary>
-		/// <param name="flags">The applicable general purpose bits flags</param>
-		/// <param name="data">
-		/// Byte array to convert
-		/// </param>
-		/// <param name="count">The number of bytes to convert.</param>
-		/// <returns>
-		/// <paramref name="data">data</paramref>converted to a string
-		/// </returns>
-		public static string ConvertToStringExt(int flags, byte[] data, int count)
-			=> (data == null)
-				? string.Empty
-				: EncodingFromFlag(flags).GetString(data, 0, count);
+		public static int SystemDefaultCodePage { get; }
 
 		/// <summary>
-		/// Convert a byte array to a string using <see cref="CodePage"/>
+		/// The system default encoding, based on <see cref="SystemDefaultCodePage"/>
 		/// </summary>
-		/// <param name="data">
-		/// Byte array to convert
-		/// </param>
-		/// <param name="flags">The applicable general purpose bits flags</param>
-		/// <returns>
-		/// <paramref name="data">data</paramref>converted to a string
-		/// </returns>
-		public static string ConvertToStringExt(int flags, byte[] data)
-			=> ConvertToStringExt(flags, data, data.Length);
+		public static Encoding SystemDefaultEncoding { get; }
 
 		/// <summary>
-		/// Convert a string to a byte array using <see cref="CodePage"/>
+		/// The encoding used for the zip archive comment. Defaults to the encoding for <see cref="CodePage"/>, since
+		/// no unicode flag can be set for it in the files.
 		/// </summary>
-		/// <param name="str">
-		/// String to convert to an array
-		/// </param>
-		/// <returns>Converted array</returns>
-		public static byte[] ConvertToArray(string str)
-			=> str == null
-			? Empty.Array<byte>()
-			: Encoding.GetEncoding(CodePage).GetBytes(str);
+		public Encoding ZipArchiveCommentEncoding
+		{
+			get => _zipArchiveCommentEncoding ?? _legacyEncoding;
+			set => _zipArchiveCommentEncoding = value;
+		}
 
 		/// <summary>
-		/// Convert a string to a byte array using <see cref="CodePage"/>
+		/// The encoding used for the ZipCrypto passwords. Defaults to <see cref="DefaultZipCryptoEncoding"/>.
 		/// </summary>
-		/// <param name="flags">The applicable <see cref="GeneralBitFlags">general purpose bits flags</see></param>
-		/// <param name="str">
-		/// String to convert to an array
-		/// </param>
-		/// <returns>Converted array</returns>
-		public static byte[] ConvertToArray(int flags, string str)
-			=> (string.IsNullOrEmpty(str))
-				? Empty.Array<byte>()
-				: EncodingFromFlag(flags).GetBytes(str);
+		public Encoding ZipCryptoEncoding
+		{
+			get => _zipArchiveCommentEncoding ?? DefaultZipCryptoEncoding;
+			set => _zipArchiveCommentEncoding = value;
```

Comment:
Is setting _zipArchiveCommentEncoding correct? (or if the intention is to share an encoding between ZipCrypto and ZipArchiveComment, maybe the backing field could have a more explanitory name?)

Reply:
Nope, thats a typo.